### ISSUE TITLE
test case not pass

### ISSUE THREAD

**Issue by flike** [pingcap/parser#40]
```
func TestSetQuery2(t *testing.T) {
   sql := "SET @master_heartbeat_period=100"
   sqlParser := parser.New()
   stmt, err := sqlParser.ParseOneStmt(sql, "", "")
   assert.Nil(t, err)
   assert.IsType(t, &ast.SetStmt{}, stmt)

   //sql := "SET @master_binlog_checksum= @@global.binlog_checksum"
   //sqlParser := parser.New()
   //stmt, err := sqlParser.ParseOneStmt(sql, "", "")
   //assert.Nil(t, err)
   //assert.IsType(t, stmt, &ast.SetStmt{})
}
```
go test -v -test.run TestSetQuery2

this case not pass,but the comment code will pass.

**Comment by winkyao** [comment]
@flike please report your golang version

**Comment by tiancaiamao** [comment]
1. Need the latest go1.11
2. Maybe you'll meet this error https://github.com/pingcap/parser/issues/4

@flike

**Comment by winkyao** [comment]
@tiancaiamao yep, flike meet the same panic error.

**Comment by flike** [comment]
@winkyao @tiancaiamao Thanks，I will try

**Comment by winkyao** [comment]
@flike If your problem is fixed after you upgrade your golang, please close this issue.

**Comment by flike** [comment]
I upgrade go version to 1.11.2, but this case still not pass. :(

**Comment by flike** [comment]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5e1902]

goroutine 1 [running]:
github.com/pingcap/parser.yyParse(0x6f8420, 0xc000140048, 0xc000140000, 0x695700)
	/root/upload/go_code/src/github.com/pingcap/parser/parser.go:8654 +0x16ab2
github.com/pingcap/parser.(*Parser).Parse(0xc000140000, 0x6b3841, 0x20, 0x0, 0x0, 0x0, 0x0, 0xc000051ee8, 0x40be28, 0x1e0, ...)
	/root/upload/go_code/src/github.com/pingcap/parser/yy_parser.go:101 +0x15f
github.com/pingcap/parser.(*Parser).ParseOneStmt(0xc000140000, 0x6b3841, 0x20, 0x0, 0x0, 0x0, 0x0, 0xc0000dc5a0, 0xc000051f88, 0x4055a0, ...)
	/root/upload/go_code/src/github.com/pingcap/parser/yy_parser.go:115 +0x7d

```

**Comment by flike** [comment]
fixed #43
